packages/core/src/MultiButton/MultiButton.stories.tsx

```diff
@@ -425,224 +427,65 @@ export const SplitButton: StoryObj<HvMultiButtonProps> = {
-              )
-            }
-          />
-        </HvMultiButton>
-        <HvMultiButton
-          variant="primarySubtle"
-          style={{ width: "232px" }}
-          split
-          disabled
-          size="xl"
-        >
-          <HvButton>{selectedOption.label}</HvButton>
-          <HvDropDownMenu
-            keepOpened={false}
-            dataList={options}
-            icon={<DropDownXS />}
-            onClick={(e, item) =>
-              setSelectedOption(
-                options.filter((option) => option.label === item.label)[0]
-              )
-            }
-          />
-        </HvMultiButton>
-        <HvMultiButton
-          variant="secondarySubtle"
-          style={{ width: "232px" }}
-          split
-          disabled
-          size="xl"
-        >
-          <HvButton>{selectedOption.label}</HvButton>
-          <HvDropDownMenu
-            keepOpened={false}
-            dataList={options}
-            icon={<DropDownXS />}
-            onClick={(e, item) =>
-              setSelectedOption(
-                options.filter((option) => option.label === item.label)[0]
-              )
-            }
-          />
-        </HvMultiButton>
+        {[
+          { variant: "primary", side: "left" },
+          { variant: "primarySubtle", side: "left" },
+          { variant: "secondarySubtle", side: "left" },
+          { variant: "primary", size: "sm", side: "right" },
+          { variant: "primarySubtle", size: "sm", side: "right" },
+          { variant: "secondarySubtle", size: "sm", side: "right" },
+          { variant: "primary", size: "lg", side: "left" },
+          { variant: "primarySubtle", size: "lg", side: "left" },
+          { variant: "secondarySubtle", size: "lg", side: "left" },
+          { variant: "primary", size: "sm", disabled: true, side: "left" },
+          {
+            variant: "primarySubtle",
+            size: "md",
+            disabled: true,
+            side: "left",
+          },
+          {
+            variant: "secondarySubtle",
+            size: "lg",
+            disabled: true,
+            side: "left",
+          },
+        ].map(({ variant, size, disabled, side }, i) => (
+          <div key={i}>
+            <HvMultiButton
+              variant={variant as HvButtonVariant}
+              size={size as HvButtonSize}
+              disabled={disabled}
+              split
+            >
+              {[
+                side === "left" && <HvButton>{selectedOption.label}</HvButton>,
+                <HvDropDownMenu
+                  dataList={options}
+                  icon={<DropDownXS />}
+                  onClick={(e, item) =>
+                    setSelectedOption(
+                      options.filter((option) => option.label === item.label)[0]
+                    )
+                  }
+                />,
+                side === "right" && <HvButton>{selectedOption.label}</HvButton>,
+              ].filter((child) => child)}
```

Comment:
I added this because of line 89 in the multi button component:

```
{split && index < React.Children.count(children) - 1 && (
```

`React.Children.count` was counting the `boolean` values leading to problems while rendering the component. Am I missing something?

Reply:
I was the one who was unaware that `Children.count` also accounted for falsy nodes. My bad.
We could `Children.toArray(children).filter(Boolean).length`, but that's too many loops 😅

Let's keep this 👍